src/plugins/discover/public/context_awareness/profile_providers/log_document_profile/profile.ts

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { createRegExpPatternFrom } from '@kbn/data-view-utils';
+import { DataTableRecord } from '@kbn/discover-utils';
+import { DocumentProfileProvider, DocumentType } from '../../profiles';
+
+export const ALLOWED_LOG_DOCUMENT_INDICES = createRegExpPatternFrom([
+  'logs',
+  'auditbeat',
+  'filebeat',
+  'winlogbeat',
+]);
+
+export const logDocumentProfileProvider: DocumentProfileProvider = {
+  profileId: 'log-document-profile',
+  profile: {},
+  resolve: ({ record }) => {
+    const isLogRecord = getIsLogRecord(record);
+
+    if (!isLogRecord) {
+      return { isMatch: false };
+    }
+
+    return {
+      isMatch: true,
+      context: {
+        type: DocumentType.Log,
+      },
+    };
+  },
+};
+
+const getIsLogRecord = (record: DataTableRecord) => {
+  return (
+    getDataStreamType(record).includes('logs') ||
+    hasFieldsWithPrefix('log.')(record) ||
+    getIndices(record).some(
+      (index) => typeof index === 'string' && ALLOWED_LOG_DOCUMENT_INDICES.test(index)
```

Comment:
When I moved the `createRegExpPatternFrom` util, I also ported in the same package the `testPatternAgainstAllowedList` utility to test a string index pattern against multiple regexp/strings:
```ts
export const ALLOWED_LOG_DOCUMENT_INDICES = [
  createRegExpPatternFrom(['logs', 'auditbeat', 'filebeat', 'winlogbeat']),
  createRegExpPatternFrom('my-company-logs'),
  'foo-logs-bar',
];

testPatternAgainstAllowedList(index, ALLOWED_LOG_DOCUMENT_INDICES)
```

Reply:
Maybe not in the short term, but as soon as we share that logic and it needs to become more flexible, we'll might have not only this single regexp to test against (maybe also user-specified indices with a setting?).
We have a maximum of 500 docs loaded at once in Discover right? I agree with "the shorter the call chain the better", although I wonder how impactful will be directly using that function